DariaCagle2(obligatory).cpp

```diff
@@ -0,0 +1,173 @@
+  int sum = kZero;
+  int digit;
+  std::cin >> number;
+  while (number != kZero) {
+    digit = number % kTen;
+    sum += digit;
+    number = (number - (number % kTen)) / kTen;
+    ++i;
+  }
+  float mean = static_cast<float>(sum) / i;
+  std::cout << sum << " " << mean << "\n";
+}
+
+void luckyTicket() {
+  std::cout << "Enter a " << kSix << "-digit number:\n";
+  int number;
+  std::cin >> number;
+  int number1 = (number - (number % kThousand)) / kThousand;
+  int number2 = number % kThousand;
+  int sum1 = kZero;
+  int digit;
+  while (number1 != kZero) {
+    digit = number1 % kTen;
+    sum1 += digit;
+    number1 = (number1 - (number1 % kTen)) / kTen;
+  }
+  int sum2 = kZero;
+  while (number2 != kZero) {
+    digit = number2 % kTen;
+    sum2 += digit;
+    number2 = (number2 - (number2 % kTen)) / kTen;
+  }
+  if (sum1 == sum2)
+    std::cout << "This is a lucky ticket!\n";
+  else {
+    std::cout << "This is not a lucky ticket!\n";
+  }
+}
+
+void reverseNumbers() {
+  __int64 number;
+  std::cout << "Enter any 32x signed number:\n";
+  std::cin >> number;
+  int digit;
+  __int64 reverseNumber = kZero;
+  if (number == abs(number)) {
+    while (number != kZero) {
+      digit = number % kTen;
+      reverseNumber = reverseNumber * kTen + digit;
+      number = (number - (number % kTen)) / kTen;
+    }
+    std::cout << reverseNumber << "\n";
+  } else {
+    number = abs(number);
+    while (number != kZero) {
+      digit = number % kTen;
+      reverseNumber = reverseNumber * kTen + digit;
+      number = (number - (number % kTen)) / kTen;
+    }
+    std::cout << -reverseNumber << "\n";
+  }
+}
+
+void sumOfOdd() {
+  int numberOfElements;
+  do {
+    std::cout << "Enter number of elements from " << kOne << " to " << kFifty
+              << ":\n";
+    std::cin >> numberOfElements;
+  } while ((numberOfElements < kOne) || (numberOfElements > kFifty));
+  int *elements = new int[numberOfElements];
+  std::cout << "Enter " << numberOfElements << " elements from " << -kSixty
+            << " to " << kNinety << ":\n";
+  size_t i = kZero;
+  while (i != numberOfElements) {
+    std::cin >> elements[i];
+    while ((elements[i] < -kSixty) || (elements[i] > kNinety)) {
+      std::cout << "Wrong range! Enter again:\n";
+      std::cin >> elements[i];
+    }
+    i++;
+  }
+  int sum = kZero;
+  for (int count = kZero; count < numberOfElements; ++count) {
+    if (elements[count] % kTwo == kOne) {
+      sum = sum + elements[count];
+    }
+  }
+  std::cout << sum << "\n";
+}
+
+int sum(int number) {
+  int digit;
+  int sum = kZero;
+  while (number != kZero) {
+    digit = number % kTen;
+    sum += digit;
+    number = (number - (number % kTen)) / kTen;
+  }
+  return sum;
+}
+
+void bestDivisor() {
+  std::cout << "Enter a positive number:\n";
+  int number;
+  std::cin >> number;
+  int bestDivisor = kZero;
+  for (int divisor = kOne; divisor <= number; divisor++) {
+    if ((number % divisor) == kZero) {
```

Comment:
```suggestion
  int bestDivisor{0};
  for (size_t divisor {0}; divisor <= number; divisor++) {
    if ((number % divisor) == 0) {
```
в текущем контексте константа kZero является лишней.
Потом счетчик цикла всегда обьявляйте с типом `size_t`

Reply:
Это я могу объяснить. Я делю инт намбер на дивайзор. Разве не будет компилятор ругаться, что здесь два разных типа? Я помню, вы что-то такое говорили.

Reply:
а да не обратил внимание. Сорри. можете оставить как есть